gnovm/stdlibs/std/emit_event.go

```diff
@@ -0,0 +1,71 @@
+package std
+
+// ref: https://github.com/gnolang/gno/pull/575
+// ref: https://github.com/gnolang/gno/pull/1833
+
+import (
+	"encoding/json"
+	"errors"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+)
+
+func X_emit(m *gno.Machine, typ string, attrs []string) {
+	eventAttrs, err := attrKeysAndValues(attrs)
+	if err != nil {
+		m.Panic(typedString(err.Error()))
+	}
+
+	pkgPath := CurrentRealmPath(m)
+	fnIdent := getPrevFunctionNameFromTarget(m, "Emit")
+
+	evt := newGnoEvent(typ, pkgPath, fnIdent, eventAttrs...)
+	ctx := m.Context.(ExecContext)
+	ctx.EventLogger.EmitEvent(evt)
+
+	str, err := json.Marshal(evt)
+	if err != nil {
+		m.Panic(typedString(err.Error()))
+	}
+
+	ctx.EmittedEvents = abci.EventString(str)
```

Comment:
I have a doubt now: does `EmittedEvents` handle only one event at a time or multiple events?

Initially, I thought you were combining the entire result, which I found good. However, now I suspect you're using this struct field as a buffer for a single event. If that's true, we should avoid using a struct field ideally. If that's not feasible or too complicated, we can keep it but consider renaming it to `lastEmittedEvent` (singular + prefixed with "last"), making it unexported, and adding a comment to explain its purpose.

Reply:
Wait, is this variable even used anywhere?

Reply:
To keep track of all events, I think modify `EmittedEvents` field by changing its type to `[]abci.EventString` and then adding the new event to the slice.

Reply:
I think modifying `EmittedEvents` here has no effect, as it's not read anywhere or even saved (you're not modifying `*ctx.EmittedEvents`). Please see my other comment for utilizing only the `EventLogger`:
https://github.com/gnolang/gno/pull/1653/files/150fe460a513dbed38cd41213ed49ab681c809cb#r1582238745

Reply:
In the PR, the events are captured from the output of the `EventLogger` anyways, so there is no need to introduce a new event store

Reply:
Removed as part of 
https://github.com/gnolang/gno/commit/7e391a90fd7a58fa2977df725cfd56a24f2b3f77